tracker/dmlc_tracker/opts.py

```diff
@@ -122,8 +144,13 @@ def get_opts(args=None):
                               ' but corresponds to archieve files that will be unziped locally,' +
                               ' You can use this option to ship python libraries.' +
                               ' Only valid in yarn jobs.'))
-    parser.add_argument('--env', action='append', default=[],
-                        help='Client and ApplicationMaster environment variables.')
```

Comment:
@rahul003 @Vikas89 I took a look at the code today and currently only `yarn.py` refers to `args.env`. The trouble is that `env` is assumed to be a list in the following line:
 https://github.com/dmlc/dmlc-core/blob/93d0122b0e864ec64b8255e56bb23b86d465460c/tracker/dmlc_tracker/yarn.py#L105-L106

So we actually do want to keep `action=append` for the `--env` argument.

Reply:
Also, the current docstring says `Comma separated list of Client and ApplicationMaster environment variables`, which clearly indicates that this argument is reserved for the YARN tracker.